ArchiSteamFarm/Core/Utilities.cs

```diff
@@ -267,6 +268,78 @@ public static class Utilities {
 		return true;
 	}
 
+	internal static async Task<bool> EnsureUpdateDirectoriesPurged(params string[] directories) {
+		if ((directories == null) || (directories.Length == 0)) {
+			throw new ArgumentNullException(nameof(directories));
+		}
+
+		foreach (string directory in directories.Where(Directory.Exists)) {
+			ASF.ArchiLogger.LogGenericInfo(Strings.UpdateCleanup);
+
+			// Apart from simple removal, we're also going to do two additional things:
+			// 1. We're going to retry if the directory is still being used
+			// 2. We're going to ensure the changes were synchronized to disk, aka fsync()
```

Comment:
I assume this is achieved by using `FileSystemWatcher`, is there a guarantee for all of supported platforms/OSes that changes are synced when we've got an event?

Reply:
No guarantee, but it's better than arbitrary delay or not waiting at all. I've tested this to work on Linux and Windows at the very least.

Considering this works on underlying OS implementation. e.g. `inotify` on Linux, we can assume pretty correct that the event of deletion won't fire if file is still being deleted from the disk.